src/environment.rs

```diff
@@ -534,14 +535,15 @@ fn remove_old_python_distributions(
             .dist_info
             .strip_prefix(site_package_path)
             .expect("the dist-info path must be a sub-path of the site-packages path");
-        rip::uninstall::uninstall_distribution(&prefix.root().join(site_package_path), relative_dist_info)
-            .into_diagnostic()
-            .with_context(|| format!("could not uninstall python package {}-{}. Manually remove the `.pixi/env` folder and try again.", &python_package.name, &python_package.version))?;
 
         // HACK: Also remove the HASH file that pixi writes. Ignore the error if its there. We
         // should probably actually add this file to the RECORD.
         let _ = std::fs::remove_file(prefix.root().join(&python_package.dist_info).join("HASH"));
 
+        rip::uninstall::uninstall_distribution(&prefix.root().join(site_package_path), relative_dist_info)
```

Comment:
This does look like some code duplication, can't uninstalling be moved into a function?

Reply:
Yes! Done :)